File: backend/src/main/java/com/ambrosia/ambrosia/services/MaterialService.java
package com.ambrosia.ambrosia.services;

import com.ambrosia.ambrosia.models.Material;
import com.ambrosia.ambrosia.models.Profesional;
import com.ambrosia.ambrosia.models.dto.MaterialDTO;
import com.ambrosia.ambrosia.repository.MaterialRepository;
import com.ambrosia.ambrosia.repository.ProfesionalRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;
import java.util.stream.Collectors;

@Service
@RequiredArgsConstructor
public class MaterialService {

    private final MaterialRepository materialRepository;
    private final ProfesionalRepository profesionalRepository;

    public MaterialDTO createMaterial(MaterialDTO materialDTO) {
        Profesional profesional = profesionalRepository.findById(materialDTO.getProfesionalId())
                .orElseThrow(() -> new RuntimeException("Profesional no encontrado"));

        Material material = Material.builder()
                .titulo(materialDTO.getTitulo())
                .contenidoHtml(materialDTO.getContenidoHtml())
                .tipo(materialDTO.getTipo())
                .profesional(profesional)
                .build();

        Material savedMaterial = materialRepository.save(material);
        return convertToDTO(savedMaterial);
    }

    public List<MaterialDTO> getAllMaterials() {
        return materialRepository.findAll().stream()
                .map(this::convertToDTO)
                .collect(Collectors.toList());
    }

    public List<MaterialDTO> getMaterialsByProfesionalId(Long profesionalId) {
        return materialRepository.findByProfesionalId(profesionalId).stream()
                .map(this::convertToDTO)
                .collect(Collectors.toList());
    }

    public Optional<MaterialDTO> getMaterialById(Long id) {
        return materialRepository.findById(id)
                .map(this::convertToDTO);
    }

    public MaterialDTO updateMaterial(Long id, MaterialDTO materialDTO) {
        Material material = materialRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Material no encontrado"));

        material.setTitulo(materialDTO.getTitulo());
        material.setContenidoHtml(materialDTO.getContenidoHtml());
        material.setTipo(materialDTO.getTipo());

        Material updatedMaterial = materialRepository.save(material);
        return convertToDTO(updatedMaterial);
    }

    public void deleteMaterial(Long id) {
        materialRepository.deleteById(id);
    }

    private MaterialDTO convertToDTO(Material material) {
        return MaterialDTO.builder()
                .id(material.getId())
                .titulo(material.getTitulo())
                .contenidoHtml(material.getContenidoHtml())
                .tipo(material.getTipo())
                .fechaCreacion(material.getFechaCreacion())
                .profesionalId(material.getProfesional().getId())
                .nombreProfesional(material.getProfesional().getUsuario().getNombre())
                .build();
    }
}

File: frontend/src/modules/profesional/services/material.service.ts
import axiosInstance from '@utils/axiosInstance';
import type { Material } from '../types/material.types';

const API_URL = '/materials';

export const createMaterial = async (material: Omit<Material, 'id' | 'fechaCreacion' | 'nombreProfesional'>): Promise<Material> => {
    const response = await axiosInstance.post<Material>(API_URL, material);
    return response.data;
};

export const getAllMaterials = async (): Promise<Material[]> => {
    const response = await axiosInstance.get<Material[]>(API_URL);
    return response.data;
};

export const getMaterialsByProfesionalId = async (profesionalId: number): Promise<Material[]> => {
    const response = await axiosInstance.get<Material[]>(`${API_URL}/profesional/${profesionalId}`);
    return response.data;
};

export const getMaterialById = async (id: number): Promise<Material> => {
    const response = await axiosInstance.get<Material>(`${API_URL}/${id}`);
    return response.data;
};

export const updateMaterial = async (id: number, material: Omit<Material, 'id' | 'fechaCreacion' | 'nombreProfesional'>): Promise<Material> => {
    const response = await axiosInstance.put<Material>(`${API_URL}/${id}`, material);
    return response.data;
};

export const deleteMaterial = async (id: number): Promise<void> => {
    await axiosInstance.delete(`${API_URL}/${id}`);
};
